graspologic/embed/ase.py

```diff
@@ -158,4 +167,64 @@ def fit(self, graph, y=None):
 
         self.n_features_in_ = len(A)
         self._reduce_dim(A)
+
+        # for out-of-sample
+        inv_eigs = np.diag(1 / self.singular_values_)
+        self._pinv_left = self.latent_left_ @ inv_eigs
+        if self.latent_right_ is not None:
+            self._pinv_right = self.latent_right_ @ inv_eigs
+
+        self.is_fitted_ = True
         return self
+
+    def transform(self, X):
+        """
+        Obtain latent positions from an adjacency matrix or matrix of out-of-sample vertices.
+        For more details on transforming out-of-sample vertices, see [2].
+
+        Parameters
+        ----------
+        X : array_like or tuple, shape (n_vertices, n_vertices) or (n_oos_vertices, n_vertices).
```

Comment:
do we care that this is called `graph` in fit?

Reply:
well, this doesn't eat a networkX graph. It's also not technically the adjacency matrix for a graph, since if it's shape (n_oos_vertices, n_vertices) then it isn't necessarily square. I can rename it if we want to line them up, though

Reply:
fc5c944